components/scream/cime_config/namelist_defaults_scream.xml

```diff
@@ -0,0 +1,210 @@
+                           one or more selectors.
+        Selector: An XML attribute that checks a property of the case. See the documentation
+                  above the <selectors> element above for more details.
+        Metadata: An XML attribute that represents metadata for an input parameter
+
+      Rules:
+        1) Parameters can have any number of parameter elements. A parameter element
+           with no selectors, which means it controls the default param value, must
+           come before parameter elements with selectors. If multiple selection elements
+           are matched, then the match the occurs last takes precendence.
+        2) The first parameter element encounter for a parameter will define the default metadata
+           for that parameter. All subsequent parameter elements will assume the same
+           metadata as the first unless an explicit override occurs.
+        3) For a parameter whose elements all have selectors, if none of the selectors
+           are matched, then the parameter will be omitted from the inputs.
+        4) Parameter types will be inferred. You can override the inferred type
+           via the 'type' metadata attribute. Types are:
+             logical: will be inferred from any case of true/false
+             integer: will be inferred from any sequence of 0-9
+             real   : will be inferred from any numeric expression with a decimal point or scientific notation
+             string : everything else
+        5) A parameter can be set to any value of the correct type unless
+           the 'valid_values' metadata attribute is set. If desired, this
+           metadata should be set up as a comma-separated list of values.
+           TODO support numeric checks (<, >, ...)??
+        6) Parameter metadata can be overriden in subsequent parameter elements, but
+           by default, metadata will be the same as the first parameter element. The
+           final metadata for a parameter will reflect what's in the last matching
+           parameter element. User changes to the case XML file must be compatible
+        7) You can inline case values into parameter values via this syntax:
+           <param>MY GRID IS ${ATM_GRID}</param>
+           This would result in parameter 'param' having the value 'MY GRID IS ne4ne4'
+        8) Any parameter value with a comma will be assumed to be a list.
+           TODO allow this behavior to be overridden?
+        9) Every value in a list should be of the same type
+       10) All local changes made to $case/namelist_scream.xml must conform
+           to the structure and metadata derived from this file.
+
+  -->
+
+  <file name="data/scream_input.yaml" format="yaml">
+    <Atmosphere__Driver>
+      <Atmosphere__Processes>
+        <Number__of__Entries>6</Number__of__Entries>
+        <Schedule__Type>Sequential</Schedule__Type>
+        <Process__0>
+          <Process__Name>Homme</Process__Name>
```

Comment:
dumb question, but why are all the underscores double-underscores in this file? Should this be added to the rules?

Reply:
That's a workaround to the fact that xml does not support spaces in the element tag. Two underscores signal buildnml that it should replace them with a space when generating the yaml file. Adding a single underscore would have been problematic, since it is something we _could_ get in a valid name.

Reply:
Well, I think it's cleaner to change our formatting in yaml files, from `Atmosphere Driver` to `atmosphere_driver`, so we don't need any shenanigan like that...

Reply:
Another downstream PR, right?

Reply:
Yes, separate one. But coming soon, so I don't think documenting the "__" vs " " issue is necessary.

Reply:
A temporary workaround. Our scream_input.yaml took advantage of the fact that YAML allows keys to have whitespace in them, e.g. "My Var". Unfortunately, in XML, "<My Var>" is not allowed, so we have to jump through some hoops for everything to work.